elasticgraph-indexer/lib/elastic_graph/indexer/datastore_indexing_router.rb

```diff
@@ -291,6 +295,7 @@ def source_event_versions_in_index(operations)
 
         if failures.empty?
           client_names_and_results.each_with_object(_ = {}) do |(client_name, _success_or_failure, results), accum|
+            results.is_a?(Hash) or raise
```

Comment:
```suggestion
```

Reply:
I think we can add type annotation the other PR to overwrite the type of the variable.